### ISSUE TITLE
HDR mode reports much higher brightness than noticeable

### ISSUE THREAD

**Issue by Ambroos** [alin23/Lunar#86]
# Issue details
- Mac device where Lunar is installed (Macbook Pro 2019, iMac, Mac Mini, Hackintosh etc.):
    *write response here*

- Monitor model (LG UltraFine 5K, DELL P2715Q etc.):
    *write response here*

- Monitor connection to the Mac device (HDMI-to-USB-C, USB-C-to-USB-C, miniDisplayPort-to-DisplayPort etc.):
    *write response here*

- Using an USB Docking Station or Hub:
    *write yes or no here*

- Lunar mode used (check it in the top-right corner of the Lunar interface)
    *Sync, Location or Manual*

- (only if you know how to compile a C program) Does this utility work for you? https://github.com/kfix/ddcctl
    *optional response*

# Issue description:
I have Lunar set to sync, contrast locked at 75 (native value for my monitor), brightness between 0 and 100. Brightness offset is at -42 to match the laptop display and monitor.

With my laptop brightness set 3 'blocks'/keyboard presses below maximum, monitor brightness is set to 34 which is perfect. Without changing the brightness, if I play the video linked below in QuickTime, monitor brightness goes up to 75. Brightness on the laptop display remains the same. I suspect it has something to do with either HDR metadata on the video file or that it's a H265 file.

Video (download source file to reproduce): https://drive.google.com/file/d/1uL51QxNnGIpnI7NkAUb3UJE0jYAKu_Yo/view?usp=sharing

Any clue what this could be?

**Comment by alin23** [comment]
You probably have Quicktime listed in the app exceptions list of Lunar:
![image](https://user-images.githubusercontent.com/3819725/77222787-372d4400-6b5f-11ea-8196-a31c7ff775ab.png)

**Comment by Ambroos** [comment]
@alin23 Nope! No exceptions! I think the HDR metadata might change something about how macOS handles display brightness, maybe?

**Comment by alin23** [comment]
I'm really not sure about the HDR thing. Can you check the output of `ioreg -c AppleBacklightDisplay | rg -o '"[^"]*brightness[^}]+"value"=[0-9]+}'` before and after starting the HDR video playback?

That will show you parameters of the built-in display brightness. Also you need ripgrep for that in case you don't have it installed: `brew install ripgrep`

**Comment by Ambroos** [comment]
The values do change!

Before:

```
"linear-brightness-probe"={"min"=0,"max"=1013,"value"=279}
"linear-brightness"={"min"=0,"max"=1013,"value"=279}
"brightness-fade"={"min"=0,"max"=1023,"value"=0}
"usable-linear-brightness"={"min"=10,"max"=1013,"value"=279}
"brightness"={"max"=1024,"min"=0,"value"=768}
"brightness-probe"={"min"=0,"max"=1024,"value"=768}
```

After:
```
"linear-brightness-probe"={"min"=0,"max"=1013,"value"=279}
"linear-brightness"={"min"=0,"max"=1013,"value"=558}
"brightness-fade"={"min"=0,"max"=1023,"value"=0}
"usable-linear-brightness"={"min"=10,"max"=1013,"value"=558}
"brightness"={"max"=1024,"min"=0,"value"=907}
"brightness-probe"={"min"=0,"max"=1024,"value"=768}
```

They return to before once I close QuickTime.

**Comment by alin23** [comment]
Well I guess that's macOS's fault ¯\\_(ツ)_/¯ 
Do you have an idea of how Lunar should treat this case?

**Comment by alin23** [comment]
Indeed, I could reproduce this too with an MP4 HDR video. It's weird because the built-in display brightness doesn't seem to change but the value reported by IOKit (which is actually what Lunar uses) really changes.

I'll try to see if there is a way to read the brightness unaffected by HDR, but I doubt I'll find anything.

Thanks for reporting this weird issue, I wouldn't have expected this.

**Comment by alin23** [comment]
This should be fixed in Lunar 4.1.4

Having 10.15 as the minimum macOS version allowed me to start using the private DisplayServices framework to read the builtin brightness which doesn't fluctuate when playing HDR content.
